eugl/mndwi.py

```diff
@@ -0,0 +1,381 @@
+
+    # get platform name
+    md = yaml.load(fid[granule + "/METADATA/CURRENT"][()], Loader=yaml.FullLoader)
+    platform_id = md["source_datasets"]["platform_id"]
+
+    # Get the 1st (lowerVal_arr) and 99th (upperVal_arr) percentikes
+    # for the each of the products. These arrays are very useful
+    # for creating MNDWI images, but this code will be removed once
+    # testing is complete.
+    lowerVal_arr = np.zeros([len(products)], order="C", dtype=np.float32)
+    upperVal_arr = np.zeros([len(products)], order="C", dtype=np.float32)
+
+    # store mndwi-based products into a group
+    mndwi_grp = h5_fid.create_group("mndwi")
+
+    for i, prod in enumerate(products):
+
+        # search the h5 groups & get paths to the green and swir bands
+        green_path, swir_path = get_mndwi_bands(granule, platform_id, prod, paths)
+
+        green_ds = grp[green_path]
+        green_im = green_ds[:]
+        geobox = GriddedGeoBox.from_dataset(green_ds)
+        spatial_res = abs(geobox.transform.a)
+        nodata = green_ds.attrs["no_data_value"]
+
+        if platform_id.startswith("SENTINEL_2"):
+            # we need to upscale the swir band
+            swir_ds = grp[swir_path]
+            swir_im = reproject_array_to_array(
+                src_img=swir_ds[:],
+                src_geobox=GriddedGeoBox.from_dataset(swir_ds),
+                dst_geobox=geobox,
+                src_nodata=swir_ds.attrs["no_data_value"],
+                dst_nodata=nodata,
+                resampling=Resampling.bilinear,
+            )
+
+        else:
+            swir_im = grp[swir_path][:]
+
+        # ------------------------- #
+        #  Compute mndwi via tiles  #
+        #   and save tiles to h5    #
+        # ------------------------- #
+        nRows, nCols = green_im.shape
+        tiles = generate_tiles(
+            samples=nRows, lines=nCols, xtile=green_ds.chunks[1], ytile=green_ds.chunks[0]
+        )
+
+        # create mndwi dataset
+        mndwi_ds = mndwi_grp.create_dataset(
+            f"mndwi_image_{prod}",
+            shape=green_im.shape,
+            dtype="float32",
+            compression="lzf",
+            chunks=green_ds.chunks,
+            shuffle=True,
+        )
+
+        # create h5 attributes
+        desc = "MNDWI ({0} m) derived with {1} and {2} ({3} reflectances)".format(
+            int(spatial_res),
+            psplit(green_path)[-1],
+            psplit(swir_path)[-1],
+            prod,
+        )
+
+        attrs = {
+            "crs_wkt": geobox.crs.ExportToWkt(),
+            "geotransform": geobox.transform.to_gdal(),
+            "no_data_value": nodata,
+            "granule": granule,
+            "description": desc,
+            "n_lines": nRows,
+            "n_samples": nCols,
+            "platform": platform_id,
+            "spatial_resolution": spatial_res,
+        }
+
+        # add attrs to dataset
+        for key in attrs:
+            mndwi_ds.attrs[key] = attrs[key]
```

Comment:
Would be ideal to include some additional attrs.  You can use this [function](https://github.com/GeoscienceAustralia/wagl/blob/develop/wagl/hdf5/__init__.py#L62) to attach them, which also allows to pass through any other attributes that you also want attached.  That will forgo the need to loop through the attributes yourself.

Reply:
It's specifically for defining IMAGE class datasets.

Reply:
What other additional attrs should I include?